src/Raven.Server/Documents/Sharding/Subscriptions/SubscriptionConnectionsStateOrchestrator.cs

```diff
@@ -170,7 +170,8 @@ public void DisposeWorkers()
         {
             try
             {
-                w.Value.Dispose();
+                // this code can be called from sharded worker itself (AbstractSubscriptionWorker._subscriptionTask)
+                w.Value.Dispose(waitForSubscriptionTask: false);
```

Comment:
why not to wait?

Reply:
as I wrote in the comment,

dispose method can be called from here https://github.com/ravendb/ravendb/blob/d7a22efd86ca17183176963f63dcce05157b5e5a/src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionWorker.cs#L127
which is the called in the task that we await in the dispose
https://github.com/ravendb/ravendb/blob/d7a22efd86ca17183176963f63dcce05157b5e5a/src/Raven.Client/Documents/Subscriptions/AbstractSubscriptionWorker.cs#L170

Reply:
done